### PR TITLE
Add pre-commit plugin .js files compiling and building

### PR DESCRIPTION
**Pull request by tjzel**
## Summary

If you forgor to build the plugin it will do it for you. But pls don't forgor.

## Test plan

Forgor on purpose and see angry message.

### CODE REVIEW

**Review by WoLewicki — APPROVED** [review]
forgor

File: .husky/pre-commit [hunk 1]
```diff
@@ -3,3 +3,14 @@
 
 yarn lint-staged
 yarn type:check
+
+# this automatically builds plugin js files if you forgor
+! git diff HEAD --name-only | grep plugin
+if [ $? -eq 1 ]; then
```
Comment (tomekzaw) [inline .husky/pre-commit:9]: Is it recommended to use return code instead of embedding the command here directly?

File: .husky/pre-commit [hunk 2]
```diff
@@ -3,3 +3,14 @@
...
+  yarn plugin
+  ! git status -u | grep -E 'plugin/build/plugin.js|plugin/build/plugin.js.map'
```
Comment (tomekzaw) [inline .husky/pre-commit:11]: 1. Why not `-e`?
2. `.` will match any character, we should either escape the dots or match just `plugin/build` (this also resolves 1.).

**Review by piaskowyk — APPROVED** [review]
👀

File: .husky/pre-commit [hunk 3]
```diff
@@ -3,3 +3,14 @@
...
+  if [ $? -eq 1 ]; then
+    echo "\n[Git hook] You forgot to build plugin and I had to do it for you!!!\n"
```
Comment (tomekzaw) [inline .husky/pre-commit:13]: Let's make this error more serious (funny PR descriptions are still fine 😄 ).

**Review by kacperkapusciak — APPROVED** [review]

**Review by tomekzaw — CHANGES_REQUESTED** [review]

File: .husky/pre-commit [hunk 4]
```diff
@@ -3,3 +3,14 @@
 
 yarn lint-staged
 yarn type:check
+
+# this automatically builds plugin js files if you forgor
+! git diff HEAD --name-only | grep plugin
```
[the remark this answers is not in the document]
Comment (tjzel) [reply .husky/pre-commit:8]: `diff-index` is low level and compares mainly metadata of a file blob. `diff` is more general thus slower according to [this article](https://stackoverflow.com/questions/24197606/whats-the-difference-between-git-diff-and-gif-diff-index). I guess for our sake `diff-index` is better but actual performance gain is negligible.

File: .husky/pre-commit (hunk 1 above)
Comment (tjzel) [reply .husky/pre-commit:9]: People have different [opinions](https://unix.stackexchange.com/questions/152966/should-i-embed-a-command-in-a-conditional-or-test-its-return-code) but I think it's more transparent in this form.

File: .husky/pre-commit (hunk 2 above)
Comment (tjzel) [reply .husky/pre-commit:11]: actually -e is not needed anyway and I actually forgot escaping

**Review by tomekzaw — APPROVED** [review]

**Pull request merged**